deposit-messaging/src/test/java/org/dataconservancy/pass/deposit/messaging/service/SubmissionProcessorIT.java

```diff
@@ -0,0 +1,170 @@
+ */
+package org.dataconservancy.pass.deposit.messaging.service;
+
+import org.dataconservancy.nihms.builder.fs.FcrepoModelBuilder;
+import org.dataconservancy.nihms.builder.fs.PassJsonFedoraAdapter;
+import org.dataconservancy.pass.client.PassClient;
+import org.dataconservancy.pass.deposit.messaging.Condition;
+import org.dataconservancy.pass.model.Deposit;
+import org.dataconservancy.pass.model.PassEntity;
+import org.dataconservancy.pass.model.Repository;
+import org.dataconservancy.pass.model.RepositoryCopy;
+import org.dataconservancy.pass.model.Submission;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toSet;
+import static org.dataconservancy.pass.model.Deposit.DepositStatus.ACCEPTED;
+import static org.dataconservancy.pass.model.RepositoryCopy.CopyStatus.COMPLETE;
+import static org.dataconservancy.pass.model.Submission.AggregatedDepositStatus.NOT_STARTED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Elliot Metsger (emetsger@jhu.edu)
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(properties = { "spring.jms.listener.auto-startup=false" })
+public class SubmissionProcessorIT {
+
+    private static final String SUBMISSION_RESOURCES = "SubmissionProcessorIT.json";
+
+    private static final String EXPECTED_REPO_NAME = "JScholarship";
+
+    private Submission submission;
+
+    @Autowired
+    @Qualifier("submissionProcessor")
+    private SubmissionProcessor underTest;
+
+    @Autowired
+    private PassClient passClient;
+
+    /**
+     * Populates Fedora with a Submission, as if it was submitted interactively by a user of the PASS UI.
+     *
+     * @throws Exception
+     */
+    @Before
+    public void createSubmission() throws Exception {
+        PassJsonFedoraAdapter passAdapter = new PassJsonFedoraAdapter();
+
+        // Upload sample data to Fedora repository to get its Submission URI.
+        InputStream is = this.getClass().getResourceAsStream(SUBMISSION_RESOURCES);
+        assertNotNull("Unable to resolve classpath resource " + SUBMISSION_RESOURCES, is);
+
+        HashMap<URI, PassEntity> uriMap = new HashMap<>();
+        URI submissionUri = passAdapter.jsonToFcrepo(is, uriMap);
+        is.close();
+
+        // Find the Submission entity that was uploaded
+        for (URI key : uriMap.keySet()) {
```

Comment:
Not a big deal, but you should be able to get the PassEntity from the list using the URI key and cast it directly to a Submission.  The URI keys are updated to use the real Fedora URLs at the end of the uploading process.